R/SpectraWrapper.R

```diff
@@ -1,7 +1,168 @@
-#' @title SpectraWrapper
+#' @title RunSpectra
 #' @description Wrapper function for the Spectra package
 #' @param seuratObj Input Seurat object
+#' @param seuratToAdataDir Directory to house files used in generating Ann Data
+#' @param assayName Assay used for generating Ann Data
+#' @param inputJSONfile Input Seurat object
+#' @param immportfile Input Seurat object
 
-SpectraWrapper <- function(seuratObj) {
-  print("Future site of Spectra Wrapper!")
+RunSpectra <- function(seuratObj, seuratToAdataDir, assayName, inputJSONfile, immportfile){
+
+  SeuratToAData(seuratObj, seuratToAdataDir, assayName)
+  gex_datafile = paste0(seuratToAdataDir, "/GEX.h5")
+  metadata_file = paste0(seuratToAdataDir, "/metadata.tsv")
+  genesets_json = paste0(seuratToAdataDir, "/genesets.json")
+  variable_features_file = paste0(seuratToAdataDir, "varfeats.csv")
+
+  ProduceGenesetJSONfile(inputJSONfile, immportfile, genesets_json)
+
+  #copy run_Spectra.py in inst/scripts and supply custom arguments
+  str <- readr::read_file(system.file("scripts/run_Spectra.py", package = "CellMembrane"))
+  script <- tempfile()
+  readr::write_file(str, script)
+
+  # run_Spectra(features_file, metadata_file, gex_datafile, genesets_json)
+  newstr <- paste0("run_Spectra(features_file = '", variable_features_file,
+                   "', gex_datafile = '", gex_datafile,
+                   "', metadata_file = '", metadata_file,
```

Comment:
TODO: we can probably assume that the `use_cell_types` parameter for spectra will be fixed to TRUE, so we should parse the genesets json file and check that the cell types in the metadata at least contain a common cell type.